symengine/tests/printing/test_printing.cpp

```diff
@@ -196,87 +197,74 @@ void test_printing()
-    assert(r1->__str__() == "Subs(Derivative(f(y, x), x), (x), (x + y))");
+    REQUIRE(r1->__str__() == "Subs(Derivative(f(y, x), x), (x), (x + y))");
 }
 
-void test_matrix()
+TEST_CASE("test_matrix(): printing", "[printing]")
 {
     DenseMatrix A = DenseMatrix(2, 2, {integer(1), integer(0), integer(0),
         integer(1)});
-    assert(A.__str__() == "[1, 0]\n[0, 1]\n");
+    REQUIRE(A.__str__() == "[1, 0]\n[0, 1]\n");
 }
 
-void test_univariate_polynomial()
+TEST_CASE("test_univariate_polynomial(): printing", "[printing]")
 {
     RCP<const Basic> p;
     RCP<const Symbol> x = symbol("x");
 
     p = univariate_polynomial(x, 0, {{0, 0}});
-    assert(p->__str__() == "0");
+    REQUIRE(p->__str__() == "0");
     //std::cout<<p->__str__()<<std::endl;
     p = univariate_polynomial(x, 0, {{0, 1}});
-    assert(p->__str__() == "1");
+    REQUIRE(p->__str__() == "1");
     //std::cout<<p->__str__()<<std::endl;
     p = univariate_polynomial(x, 1, {{1, 1}});
-    assert(p->__str__() == "x");
+    REQUIRE(p->__str__() == "x");
     //std::cout<<p->__str__()<<std::endl;
     p = univariate_polynomial(x, 1, {{0, 1}, {1, 2}});
-    assert(p->__str__() == "2*x + 1");
+    REQUIRE(p->__str__() == "2*x + 1");
     //std::cout<<p->__str__()<<std::endl;
     p = univariate_polynomial(x, 1, {{0, -1}, {1, 2}});
-    assert(p->__str__() == "2*x - 1");
+    REQUIRE(p->__str__() == "2*x - 1");
     //std::cout<<p->__str__()<<std::endl;
     p = univariate_polynomial(x, 0, {{0, -1}});
-    assert(p->__str__() == "-1");
+    REQUIRE(p->__str__() == "-1");
     //std::cout<<p->__str__()<<std::endl;
     p = univariate_polynomial(x, 1, {{1, -1}});
-    assert(p->__str__() == "-x");
+    REQUIRE(p->__str__() == "-x");
     //std::cout<<p->__str__()<<std::endl;
     p = univariate_polynomial(x, 1, {{0, -1}, {1, 1}});
-    assert(p->__str__() == "x - 1");
+    REQUIRE(p->__str__() == "x - 1");
     //std::cout<<p->__str__()<<std::endl;   
     p = univariate_polynomial(x, 2, {{0, 1}, {1, 1}, {2, 1}});
-    assert(p->__str__() == "x**2 + x + 1");
+    REQUIRE(p->__str__() == "x**2 + x + 1");
     //std::cout<<p->__str__()<<std::endl;
     p = univariate_polynomial(x, 2, {{0, 1}, {1, -1}, {2, 1}});
-    assert(p->__str__() == "x**2 - x + 1");
+    REQUIRE(p->__str__() == "x**2 - x + 1");
     //std::cout<<p->__str__()<<std::endl;
     p = univariate_polynomial(x, 2, {{0, 1}, {1, 2}, {2, 1}});
-    assert(p->__str__() == "x**2 + 2*x + 1");
+    REQUIRE(p->__str__() == "x**2 + 2*x + 1");
     //std::cout<<p->__str__()<<std::endl;
     p = univariate_polynomial(x, 2, {{1, 2}, {2, 1}});
-    assert(p->__str__() == "x**2 + 2*x");
+    REQUIRE(p->__str__() == "x**2 + 2*x");
     //std::cout<<p->__str__()<<std::endl;
     p = univariate_polynomial(x, 2, {{0, -1}, {1, -2}, {2, -1}});
     //std::cout<<p->__str__()<<std::endl;
-    assert(p->__str__() == "-x**2 - 2*x - 1");
-}
-
-int main(int argc, char* argv[])
-{
-    print_stack_on_segfault();
```

Comment:
Those functions were the worry for me, that's why I have not yet updated `eval` tests.